### ISSUE TITLE
Saw MWN_MISMATCHED_NOTIFY false positive

### ISSUE THREAD

**Issue by ctubbsii** [spotbugs/spotbugs#1949]
Saw the following code trigger a false positive:

```java
T someItem;
Set<T> setOfObjects = someSet();
try {
  doSomeWorkHere();
 } finally {
   synchronized(setOfObjects) {
     setOfObjects.remove(someItem);
     setOfObjects.notify(); // <--- This is the line reported by spotbugs with MWN_MISMATCHED_NOTIFY
   }
 }
 ```
 
This looks like a false positive to me. The description of `MWN_MISMATCHED_NOTIFY` says it happens when the lock is not obviously held. However, that synchronized block seems pretty obvious to me that the lock is held on the object.

This was observed with spotbugs-maven-plugin 4.5.3.0

**Comment by ctubbsii** [comment]
I see a comment at https://spotbugs.readthedocs.io/en/stable/detectors.html#findmismatchedwaitornotify that the detector is disabled due to too many false positives. However, that does not appear to be the case. The detector is enabled, and clearly triggered on false positives still.

**Comment by ctubbsii** [comment]
Interestingly, if I make the objects `final`, it seems to analyze correctly without the false positive.
